src/libbson/doc/bson_string_new.rst

```diff
@@ -14,7 +14,7 @@ Synopsis
 Parameters
 ----------
 
-* ``str``: A string to be copied or NULL.
+* ``str``: A string to be copied or NULL. Aborts if length with NULL terminator (``strlen(str) + 1``) exceeds ``UINT32_MAX``.
```

Comment:
```suggestion
* ``str``: A string to be copied or NULL.
```

Recommend consistently placing this line in the function "Description" section in a dedicated "warning" section:

```rst
.. warning::

  This function will abort if the length of the resulting string (including the NULL terminator) would exceed ``UINT32_MAX``.
```

Also applies to other modified doc pages in this PR.